tests/unit/preprocessor/_volume/test_volume.py

```diff
@@ -100,6 +100,16 @@ def test_volume_statistics_long(self):
         expected = np.array([1., 1., 1., 1.])
         self.assert_array_equal(result.data, expected)
 
+    def test_volume_statistics_masked_level(self):
+        """
+        Test to take the volume weighted average of a (2,3,2,2) cube
+        where the last depth level is fully masked.
+        """
+        self.grid_4d.data[:, -1, :, :] = np.ma.masked_all((2, 2, 2))
+        result = volume_statistics(self.grid_4d, 'mean')
+        expected = np.array([1., 1.])
```

Comment:
I would expect `volume_statistics` to return a cube with masked data when the input also has masked data.

Reply:
This is one spatial average value per timestep. Ocean data always contains masked values anyway and the output for this kind of operations is a non-masked value. It should not be different in this case.

Reply:
What if all values for a particular time step are masked? Are you sure that never happens?

Reply:
ya that was my previous implementation that looked a bit overkill (inc a test for it) but @sloosvel assures me that can't happen